nixos/modules/services/monitoring/timescale-prometheus.nix

```diff
@@ -0,0 +1,272 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.timescale-prometheus;
+
+in
+{
+
+  options = {
+    services.timescale-prometheus = {
+
+      enable = mkEnableOption "timescale prometheus service";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.timescale-prometheus;
+        defaultText = "pkgs.timescale-prometheus";
+        description = "The timescale-prometheus package to use.";
+      };
+
+      async-acks = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = ''
+          Ack before data is written to DB [TS_PROM_ASYNC_ACKS]
+        '';
+      };
+
+      db-connect-retries = mkOption {
```

Comment:
Again, purely a style thing, feel free to ignore 100% and then some but... I think most modules typically use a `database.` prefix for database related options, something like
```
database = {
  connect-retries = mkOption { ... };
  host = mkOption { ... };
  name = mkOption { ... };
};
```

Consistency is always nice.